### PR TITLE
ST-488: Added duplicate submission warning page when pension already updated

### PR DESCRIPTION
**Pull request by BriWak**
- [x] Unit tests passing
- [x] Coverage reached
- [x] Acceptance tests passing
- [ ] Reviewed

### CODE REVIEW

File: test/controllers/pensions/UpdatePensionProviderControllerSpec.scala [hunk 1]
```diff
@@ -425,12 +363,136 @@ class UpdatePensionProviderControllerSpec extends PlaySpec with FakeTaiPlayAppli
...
+      val sut = createSUT
+
+      when(taxAccountService.taxCodeIncomes(any(), any())(any())).
+        thenReturn(Future.successful(TaiSuccessResponseWithPayload(Seq(pensionTaxCodeIncome, empTaxCodeIncome))))
+      when(successfulJourneyCacheService.currentValue(Matchers.eq(s"$TrackSuccessfulJourney_UpdatePensionKey-$pensionId"))(any())).
+      thenReturn(Future.successful(Some("true")))
+      when(journeyCacheService.cache(Matchers.eq(cacheMap))(any())).thenReturn(Future.successful(cacheMap))
+
+      val result = sut.redirectUpdatePension(pensionId)(RequestBuilder.buildFakeRequestWithAuth("GET"))
+      status(result) mustBe SEE_OTHER
```
Comment (oc-ste) [inline test/controllers/pensions/UpdatePensionProviderControllerSpec.scala]: Dedupe test setup

File: test/controllers/pensions/UpdatePensionProviderControllerSpec.scala [hunk 2]
```diff
@@ -425,12 +363,136 @@ class UpdatePensionProviderControllerSpec extends PlaySpec with FakeTaiPlayAppli
...
+    "return BadRequest" when {
+      "there is a form validation error (standard form validation)" in {
+        val pensionId = 1
+        val updatePensionTest = createSUT
+
+        when(journeyCacheService.mandatoryValues(Matchers.anyVararg[String])(any()))
+          .thenReturn(Future.successful(Seq(pensionName, pensionId.toString)))
+
+        val result = updatePensionTest.submitDuplicateSubmissionWarning(RequestBuilder.buildFakeRequestWithAuth("POST")
+          .withFormUrlEncodedBody(YesNoChoice -> ""))
```
Comment (oc-ste) [inline test/controllers/pensions/UpdatePensionProviderControllerSpec.scala]: Dedupe test set up

File: app/views/pensions/update/duplicateSubmissionWarning.scala.html [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import includes.forms.submit
+@import uk.gov.hmrc.tai.forms.YesNoForm
+@import includes.{cancelLink, ptaHeader}
+@import uk.gov.hmrc.play.views.html.helpers.{errorSummary, form}
+@import includes.forms.inputRadioGroupStyling
+
```
Comment (oc-ste) [inline app/views/pensions/update/duplicateSubmissionWarning.scala.html]: Is this the same view as the employment journey warning?

File: app/controllers/pensions/UpdatePensionProviderController.scala [hunk 4]
```diff
@@ -239,4 +221,68 @@ class UpdatePensionProviderController @Inject()(taxAccountService: TaxAccountSer
 
       Future.successful(Ok(views.html.pensions.update.confirmation()))
   }
+
+  def redirectUpdatePension(id: Int): Action[AnyContent] = (authenticate andThen validatePerson).async {
```
Comment (oc-ste) [inline app/controllers/pensions/UpdatePensionProviderController.scala]: This stuff looks similar to the end employment journey, probably a dedupe refactor session could go on here

File: test/controllers/pensions/UpdatePensionProviderControllerSpec.scala (hunk 1 above)
Comment (BriWak) [reply test/controllers/pensions/UpdatePensionProviderControllerSpec.scala]: Deduped the fake request, journeyCache Call and taxAccountService Call now. 

Not sure if there's anything else that can be deduped?

File: test/controllers/pensions/UpdatePensionProviderControllerSpec.scala (hunk 2 above)
Comment (BriWak) [reply test/controllers/pensions/UpdatePensionProviderControllerSpec.scala]: Deduped the fake request, journeyCache Call and taxAccountService Call now.

File: app/views/pensions/update/duplicateSubmissionWarning.scala.html (hunk 3 above)
Comment (BriWak) [reply app/views/pensions/update/duplicateSubmissionWarning.scala.html]: They are different views but they now share a partial

**Pull request merged**